SoftLayer/managers/vs.py

```diff
@@ -533,10 +533,7 @@ def _generate_create_dict(
         if datacenter:
             data["datacenter"] = {"name": datacenter}
 
-        if private_vlan or public_vlan or private_subnet or public_subnet:
-            network_components = self._create_network_components(public_vlan, private_vlan,
-                                                                 private_subnet, public_subnet)
-            data.update(network_components)
+        self.get_network_components(data, kwargs, private_subnet, private_vlan, public_subnet, public_vlan)
```

Comment:
I don't like function that modify the data that gets passed into them. I'd prefer this function `get_network_componets` return `data` instead.

So this line would be:
```
network_components = self.get_network_components(kwargs, private_subnet, private_vlan, public_subnet, public_vlan)
data.update(network_components)
```
as that reads better to me.

Reply:
I did it the same lol, but many unit tests failed. I am going to check it again.

Reply:
Done.